bundles/cif-connector-graphql/src/test/java/com/adobe/cq/commerce/gui/components/common/cifproductfield/InitializerTest.java

```diff
@@ -0,0 +1,123 @@
+/*******************************************************************************
+ *
+ *    Copyright 2019 Adobe. All rights reserved.
+ *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License. You may obtain a copy
+ *    of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software distributed under
+ *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ *    OF ANY KIND, either express or implied. See the License for the specific language
+ *    governing permissions and limitations under the License.
+ *
+ ******************************************************************************/
+
+package com.adobe.cq.commerce.gui.components.common.cifproductfield;
+
+import org.apache.sling.api.resource.ValueMap;
+import org.junit.Test;
+
+import com.adobe.cq.commerce.graphql.search.CatalogSearchSupport;
+import com.adobe.cq.commerce.gui.components.common.FieldInitializerTest;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.when;
+
+public class InitializerTest extends FieldInitializerTest {
+    private Initializer initializer = new Initializer();
+
+    @Test
+    public void testDefaults() {
+        initializer.init(bindings);
+
+        assertNotNull(includedResourceSample);
+
+        ValueMap properties = includedResourceSample.getValueMap();
+        assertNotNull(properties);
+        assertEquals(Initializer.DEFAULT_FILTER, properties.get("filter", String.class));
+        assertEquals("Product ID", properties.get("emptyText", String.class));
+        assertFalse(properties.get("multiple", Boolean.class));
+        assertEquals("commerce/gui/components/common/productfield", properties.get("sling:resourceType", String.class));
+        assertEquals(PRODUCTS_BASE_PATH, properties.get("rootPath", String.class));
+        assertTrue(properties.get("forceselection", Boolean.class));
+        String pickerSrc = "/mnt/overlay/commerce/gui/content/common/cifproductfield/picker.html?root=products%2fbase%2fpath&"
+            + "filter=folderOrProduct&selectionCount=single&selectionId=id";
+        assertEquals(pickerSrc, properties.get("pickerSrc", String.class));
+    }
+
+    @Test
+    public void testSelectionId_path() {
```

Comment:
Why do you mix camelCase and snake_case here?

Reply:
OK, it's a rare exception that I happen to use underscore like this. The point would be that the part after the underscore is literally the value of selectionId parameter I'm testing for. So, there's not much theory here, just a convenience for naming a simple testcase.